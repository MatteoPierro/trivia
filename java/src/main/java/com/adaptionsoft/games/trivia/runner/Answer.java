package com.adaptionsoft.games.trivia.runner;

public class Answer {
    private static final int INCORRECT_ANSWER = 7;
    private final int answer;

    public Answer(int answer) {
        this.answer = answer;
    }

    public boolean isCorrect() {
        return answer != INCORRECT_ANSWER;
    }
}
