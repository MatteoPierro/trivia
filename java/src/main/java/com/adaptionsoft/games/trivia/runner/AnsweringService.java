package com.adaptionsoft.games.trivia.runner;

import java.util.Random;

public class AnsweringService {
    private Random random;

    public AnsweringService(Random random) {
        this.random = random;
    }

    public Answer answer() {
        int answer = random.nextInt(9);
        return new Answer(answer);
    }
 }
