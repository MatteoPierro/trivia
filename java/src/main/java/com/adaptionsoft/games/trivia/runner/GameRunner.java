
package com.adaptionsoft.games.trivia.runner;
import java.util.Random;

import com.adaptionsoft.games.uglytrivia.Game;


public class GameRunner {

	private static boolean notAWinner;

	public static void main(String[] args) {
		Random rand = new Random();
		runGame(rand);
	}

	public static void runGame(Random rand){
		runGame(rand, "Chet", "Pat", "Sue");
	}

	public static void runGame(Random rand, String... players) {
		Game aGame = new Game();

		for (String player:players) {
			aGame.add(player);

		}

		do {

			aGame.roll(rand.nextInt(5) + 1);

			if (rand.nextInt(9) == 7) {
				notAWinner = aGame.wrongAnswer();
			} else {
				notAWinner = aGame.wasCorrectlyAnswered();
			}



		} while (notAWinner);
	}
}
