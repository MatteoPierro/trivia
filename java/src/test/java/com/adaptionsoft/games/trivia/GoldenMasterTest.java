package com.adaptionsoft.games.trivia;

import com.adaptionsoft.games.trivia.runner.GameRunner;
import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.util.Random;

public class GoldenMasterTest {

	@Test
	public void can_run_a_controlled_game_test() {
		String result = runGame(1);

		System.out.println(result);
	}

	@Test
	public void can_run_a_controlled_game() {
		String result = runGame(1);

		Approvals.verify(result);
	}

	@Test
	public void can_run_controlled_game_for_multiple_seeds() {
		Integer[] seeds = {3,5};

		Approvals.verifyAll(seeds,seed -> runGame(seed));
	}

	public String runGame(int seed) {
		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
		PrintStream printStream = new PrintStream(outputStream, true);

		PrintStream oldOut = System.out;
		System.setOut(printStream);

		GameRunner.runGame(new Random(seed));

		System.setOut(oldOut);

		return outputStream.toString();
	}

	private class Players {
		private String[] players;

		public Players(String ... players) {
			this.players = players;
		}

		public String[] values() {
			return players;
		}

		@Override
		public String toString() {
			return String.join(",", players);
		}
	}

	public String runGameForSeedAndPlayers(Integer seed, Players players) {
		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
		PrintStream printStream = new PrintStream(outputStream, true);

		PrintStream oldOut = System.out;
		System.setOut(printStream);

		Random rand = new Random(seed);
		GameRunner.runGame(rand, players.players);

		System.setOut(oldOut);

		return outputStream.toString();
	}

	@Test
	public void can_run_controlled_game_for_multiple_players() throws Exception {
		Integer[] seeds = {3, 5};
		Players[] playerCombinations = new Players[] {
				new Players(),
				new Players("Chet"),
				new Players("Chet", "Jean"),
				new Players("Chet", "Jean", "Matteo", "Nelis", "Jan", "Piet"),
		};

		CombinationApprovals.verifyAllCombinations(this::runGameForSeedAndPlayers, seeds, playerCombinations);
	}
}
