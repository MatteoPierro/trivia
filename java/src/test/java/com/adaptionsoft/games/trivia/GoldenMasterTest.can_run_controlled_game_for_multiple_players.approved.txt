[3, ] => Index: 0, Size: 0 
[3, Chet] => Chet was added
They are player number 1
Chet is the current player
They have rolled a 5
Chet's new location is 5
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Chet is the current player
They have rolled a 1
Chet's new location is 6
The category is Sports
Sports Question 0
Question was incorrectly answered
Chet was sent to the penalty box
Chet is the current player
They have rolled a 4
Chet is not getting out of the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 11
The category is Rock
Rock Question 0
Question was incorrectly answered
Chet was sent to the penalty box
Chet is the current player
They have rolled a 5
Chet is getting out of the penalty box
Chet's new location is 4
The category is Pop
Pop Question 0
Answer was correct!!!!
Chet now has 2 Gold Coins.
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 5
The category is Science
Science Question 1
Answer was correct!!!!
Chet now has 3 Gold Coins.
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 8
The category is Pop
Pop Question 1
Answer was correct!!!!
Chet now has 4 Gold Coins.
Chet is the current player
They have rolled a 2
Chet is not getting out of the penalty box
Chet is the current player
They have rolled a 2
Chet is not getting out of the penalty box
Chet is the current player
They have rolled a 2
Chet is not getting out of the penalty box
Chet is the current player
They have rolled a 2
Chet is not getting out of the penalty box
Chet is the current player
They have rolled a 1
Chet is getting out of the penalty box
Chet's new location is 9
The category is Science
Science Question 2
Answer was correct!!!!
Chet now has 5 Gold Coins.
Chet is the current player
They have rolled a 4
Chet is not getting out of the penalty box
Chet is the current player
They have rolled a 3
Chet is getting out of the penalty box
Chet's new location is 0
The category is Pop
Pop Question 2
Answer was correct!!!!
Chet now has 6 Gold Coins.
 
[3, Chet,Jean] => Chet was added
They are player number 1
Jean was added
They are player number 2
Chet is the current player
They have rolled a 5
Chet's new location is 5
The category is Science
Science Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Jean is the current player
They have rolled a 1
Jean's new location is 1
The category is Science
Science Question 1
Question was incorrectly answered
Jean was sent to the penalty box
Chet is the current player
They have rolled a 4
Chet's new location is 9
The category is Science
Science Question 2
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Jean is the current player
They have rolled a 5
Jean is getting out of the penalty box
Jean's new location is 6
The category is Sports
Sports Question 0
Question was incorrectly answered
Jean was sent to the penalty box
Chet is the current player
They have rolled a 5
Chet's new location is 2
The category is Sports
Sports Question 1
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Jean is the current player
They have rolled a 1
Jean is getting out of the penalty box
Jean's new location is 7
The category is Rock
Rock Question 0
Answer was correct!!!!
Jean now has 1 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 5
The category is Science
Science Question 3
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Jean is the current player
They have rolled a 2
Jean is not getting out of the penalty box
Chet is the current player
They have rolled a 2
Chet's new location is 7
The category is Rock
Rock Question 1
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Jean is the current player
They have rolled a 2
Jean is not getting out of the penalty box
Chet is the current player
They have rolled a 2
Chet's new location is 9
The category is Science
Science Question 4
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[3, Chet,Jean,Matteo,Nelis,Jan,Piet] => 6 
[5, ] => Index: 0, Size: 0 
[5, Chet] => Chet was added
They are player number 1
Chet is the current player
They have rolled a 3
Chet's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 8
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Chet is the current player
They have rolled a 2
Chet's new location is 10
The category is Sports
Sports Question 0
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Chet is the current player
They have rolled a 5
Chet's new location is 3
The category is Rock
Rock Question 1
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 6
The category is Sports
Sports Question 1
Answer was corrent!!!!
Chet now has 5 Gold Coins.
Chet is the current player
They have rolled a 2
Chet's new location is 8
The category is Pop
Pop Question 1
Answer was corrent!!!!
Chet now has 6 Gold Coins.
 
[5, Chet,Jean] => Chet was added
They are player number 1
Jean was added
They are player number 2
Chet is the current player
They have rolled a 3
Chet's new location is 3
The category is Rock
Rock Question 0
Answer was corrent!!!!
Chet now has 1 Gold Coins.
Jean is the current player
They have rolled a 5
Jean's new location is 5
The category is Science
Science Question 0
Answer was corrent!!!!
Jean now has 1 Gold Coins.
Chet is the current player
They have rolled a 2
Chet's new location is 5
The category is Science
Science Question 1
Answer was corrent!!!!
Chet now has 2 Gold Coins.
Jean is the current player
They have rolled a 5
Jean's new location is 10
The category is Sports
Sports Question 0
Answer was corrent!!!!
Jean now has 2 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 8
The category is Pop
Pop Question 0
Answer was corrent!!!!
Chet now has 3 Gold Coins.
Jean is the current player
They have rolled a 2
Jean's new location is 0
The category is Pop
Pop Question 1
Answer was corrent!!!!
Jean now has 3 Gold Coins.
Chet is the current player
They have rolled a 3
Chet's new location is 11
The category is Rock
Rock Question 1
Answer was corrent!!!!
Chet now has 4 Gold Coins.
Jean is the current player
They have rolled a 4
Jean's new location is 4
The category is Pop
Pop Question 2
Answer was corrent!!!!
Jean now has 4 Gold Coins.
Chet is the current player
They have rolled a 4
Chet's new location is 3
The category is Rock
Rock Question 2
Question was incorrectly answered
Chet was sent to the penalty box
Jean is the current player
They have rolled a 3
Jean's new location is 7
The category is Rock
Rock Question 3
Answer was corrent!!!!
Jean now has 5 Gold Coins.
Chet is the current player
They have rolled a 4
Chet is not getting out of the penalty box
Jean is the current player
They have rolled a 4
Jean's new location is 11
The category is Rock
Rock Question 4
Answer was corrent!!!!
Jean now has 6 Gold Coins.
 
[5, Chet,Jean,Matteo,Nelis,Jan,Piet] => 6 
